### PR TITLE
fix: error request body if socket fails

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Comment by ronag** [comment]
This pool error seems flaky, I get it sometimes locally as well, not always.

File: lib/client.js [hunk 1]
```diff
@@ -164,18 +165,29 @@ class Client extends EventEmitter {
         }
         this.socket.write(body)
       } else if (body && typeof body.pipe === 'function') {
+        const cleanup = eos(this.socket, err => {
```
Comment (ronag) [inline lib/client.js:168]: I'm a little unsure here what happens if the socket unexpectedly emits `'end'`. Does `Client` somewhere detect this and call `Client.destroy(err)`?

Comment (ronag) [reply lib/client.js:168]: Created issue https://github.com/mcollina/undici/issues/21

**Comment by mcollina** [comment]
could you fix the conflicts?

**Comment by ronag** [comment]
fixed

**Pull request merged**